CMakeLists.txt

```diff
@@ -94,6 +94,14 @@ if(CONFIG_MBEDTLS_BUILTIN)
   zephyr_library_sources(library/x509write_csr.c)
   zephyr_library_sources(library/xtea.c)
 
+if(CONFIG_ARCH_POSIX AND CONFIG_ASAN AND NOT CONFIG_64BIT)
+  # i386 assembly code used in MBEDTLS does not compile with size optimization
```

Comment:
This comments seems to be backwards comparing to what's written in the PR description. My understanding is that if size optimization is enabled, then i386 assembly code contained in this file **is not** (vs does not) compiled. Whereas if we enable Address Sanitizer, we'd like to make sure it runs thru all code. That's why, when Address Sanitizer is used, we have to switch optimization level for this file.

If that's the case, then it rather be written more clearer. And if I missed it, then ... the same, description should be more clear. For example, it it truly **does not** compile, i.e. there's compilation error trying to build it, that's an apparent mbedTLS bug, is there upstream issue reference for it?

Reply:
It is actually not compiling and there is no upstream  issue for this to my knowledge